contentcuration/contentcuration/views.py

```diff
@@ -46,22 +46,57 @@ def unsupported_browser(request):
+    accessible_channel_list = Channel.objects.filter(deleted=False).filter( Q(public=True) | Q(editors=request.user) | Q(viewers=request.user))
+    accessible_channel_list = ChannelSerializer.setup_eager_loading(accessible_channel_list)
+    accessible_channel_list_serializer = ChannelSerializer(accessible_channel_list, many=True)
+
+    channel_list = accessible_channel_list.filter(Q(editors=request.user) | Q(viewers=request.user))\
+                    .exclude(id=channel.pk)\
+                    .annotate(is_view_only=Case(When(viewers=request.user,then=Value(1)),default=Value(0),output_field=IntegerField()))\
+                    .values("id", "name", "is_view_only")
+    fileformats = get_or_set_cached_constants(FileFormat, FileFormatSerializer)
+    licenses = get_or_set_cached_constants(License, LicenseSerializer)
+    formatpresets = get_or_set_cached_constants(FormatPreset, FormatPresetSerializer)
+    contentkinds = get_or_set_cached_constants(ContentKind, ContentKindSerializer)
+
+    channel_tags = ContentTag.objects.filter(channel = channel)
+    channel_tags_serializer = TagSerializer(channel_tags, many=True)
+
+    json_renderer = JSONRenderer()
+
+    return render(request, 'channel_edit.html', {"allow_edit":allow_edit,
+                                                "channel" : json_renderer.render(channel_serializer.data),
+                                                "channel_id" : channel.pk,
+                                                "channel_name": channel.name,
+                                                "accessible_channels" : json_renderer.render(accessible_channel_list_serializer.data),
+                                                "channel_list" : channel_list,
+                                                "fileformat_list" : fileformats,
+                                                 "license_list" : licenses,
+                                                 "fpreset_list" : formatpresets,
+                                                 "ckinds_list" : contentkinds,
+                                                 "ctags": json_renderer.render(channel_tags_serializer.data),
+                                                 "current_user" : json_renderer.render(CurrentUserSerializer(request.user).data)})
+
 @login_required
 @authentication_classes((SessionAuthentication, BasicAuthentication, TokenAuthentication))
 @permission_classes((IsAuthenticated,))
 def channel_list(request):
     if not check_supported_browsers(request.META['HTTP_USER_AGENT']):
         return redirect(reverse_lazy('unsupported_browser'))
 
-    channel_list = Channel.objects.filter(deleted=False, editors= request.user)
+    channel_list = Channel.objects.filter(deleted=False)\
+                    .filter(Q(editors=request.user) | Q(viewers=request.user))\
+                    .annotate(is_view_only=Case(When(viewers=request.user,then=Value(1)),default=Value(0),output_field=IntegerField()))
```

Comment:
Just for safety's sake, might be nicer to make `is_view_only` have `Value(1)` by default, and do the opposite lookup `Case(When(editors=request.user, then=Value(0)))` - that way, if something weird is happening, we don't accidentally give edit access to someone.